python/ray/data/preprocessors/vectorizer.py

```diff
@@ -9,15 +9,84 @@
 
 
 class HashingVectorizer(Preprocessor):
-    """Tokenize and hash string columns into token hash columns.
+    """Count the frequency of tokens using the
+    `hashing trick <https://en.wikipedia.org/wiki/Feature_hashing>`_.
 
-    The created columns will have names in the format ``hash_{column_name}_{hash}``.
+    This preprocessors creates ``num_features`` columns named like
+    ``hash_{column_name}_{index}``. If ``num_features`` is large enough relative to
+    the size of your vocabulary, then each column approximately corresponds to the
+    frequency of a unique token.
+
+    This preprocessor is memory efficient and quick to pickle. However, given a transformed
+    column, you can't know which tokens correspond to it. This might make it hard
+    to determine which tokens are important to your model.
+
+    .. note::
+
+        A document-term matrix is a table that describes the frequency of tokens
+        in a collection of strings. For example, the strings `"I like Python"` and `"I
+        dislike Python"` might have the document-term matrix below:
+
+        .. code-block::
+
+                    corpus_I  corpus_Python  corpus_dislike  corpus_like
+                0         1              1               1            0
+                1         1              1               0            1
+
+        To generate the matrix, you need to map each token to a unique index. For example:
+
+        .. code-block::
+
+                        token  index
+                0        I      0
+                1   Python      1
+                2  dislike      2
+                3     like      3
+
+        The problem with this approach is that the mapping and consequently memory use
+        scale linearly with the size of your vocabulary.
+
+        :class:`HashingVectorizer` circumvents this problem by computing
+        indices with a hash function: :math:`\\texttt{index} = hash(\\texttt{token})`.
 
     Args:
-        columns: The columns that will individually be hashed.
-        num_features: The size of each hashed feature vector.
-        tokenization_fn: The tokenization function to use.
-            If not specified, a simple ``string.split(" ")`` will be used.
+        columns: The columns to separately tokenize and count.
+        num_features: The number of features used to represent the vocabulary. You
+            should choose a value large enough to prevent hash collisions between
+            distinct tokens. :math:`2^{18}` is typically large enough for text-classification
+            problems.
```

Comment:
Unfortunately Datasets will not scale well to this number of columns. We would need to add support for sparse matrices for this to be effective.

Reply:
Fixed